impeller/renderer/backend/vulkan/binding_helpers.cc

```diff
@@ -0,0 +1,225 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "impeller/renderer/backend/vulkan/binding_helpers.h"
+#include "impeller/renderer/backend/vulkan/command_buffer_vk.h"
+#include "impeller/renderer/backend/vulkan/command_encoder_vk.h"
+#include "impeller/renderer/backend/vulkan/command_pool_vk.h"
+#include "impeller/renderer/backend/vulkan/compute_pipeline_vk.h"
+#include "impeller/renderer/backend/vulkan/context_vk.h"
+#include "impeller/renderer/backend/vulkan/sampler_vk.h"
+#include "impeller/renderer/backend/vulkan/texture_vk.h"
+#include "impeller/renderer/backend/vulkan/vk.h"
+#include "impeller/renderer/command.h"
+#include "impeller/renderer/compute_command.h"
+
+namespace impeller {
+
+static bool BindImages(const Bindings& bindings,
+                       Allocator& allocator,
+                       const std::shared_ptr<CommandEncoderVK>& encoder,
+                       vk::DescriptorSet& vk_desc_set,
+                       std::vector<vk::DescriptorImageInfo>& images,
+                       std::vector<vk::WriteDescriptorSet>& writes) {
+  for (const auto& [index, data] : bindings.sampled_images) {
+    auto texture = data.texture.resource;
+    const auto& texture_vk = TextureVK::Cast(*texture);
+    const SamplerVK& sampler = SamplerVK::Cast(*data.sampler.resource);
+
+    if (!encoder->Track(texture) ||
+        !encoder->Track(sampler.GetSharedSampler())) {
+      return false;
+    }
```

Comment:
TODO: we're currently calling track twice on each texture.

Reply:
FWIW, tracking the same resource multiple times is safe.